vendorpromo/forms.py

```diff
@@ -1,9 +1,27 @@
 from django import forms
+from django.contrib.sites.models import Site
+from django.db.models import TextChoices
 from django.forms import modelformset_factory
 from django.utils.translation import ugettext as _
 
 from vendorpromo.models import Promo
 
+class SupportedPromoProcessor(TextChoices):
+    PROMO_CODE_BASE = ("base.PromoProcessorBase", _("Default Processor"))
+    VOUCHERY = ("vouchery.VoucheryProcessor", _("Vouchery.io"))
+
+class PromoProcessorForm(forms.Form):
+    promo_processor = forms.CharField(label=_("Processor"), widget=forms.Select(choices=SupportedPromoProcessor.choices))
+
+    def __init__(self, *args, **kwargs):
```

Comment:
seems like this init is a repeat of your earlier declaration of the promo_processor field; if you change the label from "processor" to "promo processor", it should achieve the same result

Reply:
it overrides the forms label and widget type

Reply:
you are right I will remove it